baybe/recommenders/pure/bayesian/botorch.py

```diff
@@ -314,23 +456,40 @@ def _recommend_hybrid(
 
         return rec_exp
 
-    @override
-    def __str__(self) -> str:
-        fields = [
-            to_string("Surrogate", self._surrogate_model),
-            to_string(
-                "Acquisition function", self.acquisition_function, single_line=True
-            ),
-            to_string("Compatibility", self.compatibility, single_line=True),
-            to_string(
-                "Sequential continuous", self.sequential_continuous, single_line=True
-            ),
-            to_string("Hybrid sampler", self.hybrid_sampler, single_line=True),
-            to_string(
-                "Sampling percentage", self.sampling_percentage, single_line=True
-            ),
-        ]
-        return to_string(self.__class__.__name__, *fields)
+    def _optimize_continuous_subspaces(
```

Comment:
@AVHopp @Scienfitz: Just wanted to raise awareness. I've refactored @Waschenbacher's original version quite significantly, but the nice outcome of this is that we now have this useful and general helper that can be used to optimize over arbitrary non-connected subspaces. Thumbs up once you saw it and I'll resolve

Reply:
Can we potentially leverage this for Discrete Searchspaces or some sort of hybrid recommendation as well?

Reply:
Yes, the use is by no means limited to cardinality constraints. It's basically the baybe equivalent to botorchs `optimize_acqf_mixed`, if you will.